packages/adapter-drizzle-pg/src/index.ts

```diff
@@ -0,0 +1,264 @@
+ *  expires_at: integer("expires_at"),
+ *  token_type: text("token_type"),
+ *  scope: text("scope"),
+ *  id_token: text("id_token"),
+ *  session_state: text("session_state"),
+ * }, (account) => ({
+ *   _: primaryKey(account.provider, account.providerAccountId)
+ * }))
+ *
+ * export const sessions = pgTable("sessions", {
+ *  userId: text("userId").notNull().references(() => users.id, { onDelete: "cascade" }),
+ *  sessionToken: text("sessionToken").notNull().primaryKey(),
+ *  expires: integer("expires").notNull(),
+ * })
+ *
+ * export const verificationTokens = pgTable("verificationToken", {
+ *  identifier: text("identifier").notNull(),
+ *  token: text("token").notNull(),
+ *  expires: integer("expires").notNull()
+ * }, (vt) => ({
+ *   _: primaryKey(vt.identifier, vt.token)
+ * }))
+ *
+ * const pool = new Pool({
+ *   connectionString: "YOUR_CONNECTION_STRING"
+ * });
+ *
+ * export const db = drizzle(pool);
+ *
+ * migrate(db, { migrationsFolder: "./drizzle" })
+ *
+ * ```
+ *
+ **/
+export function DrizzleAdapter(client: typeof db): Adapter {
+  return {
+    createUser: async (data) => {
+      return client
+        .insert(users)
+        .values({ ...data, id: "123" })
+        .returning()
+        .then(res => res[0])
+    },
+    getUser: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data))
+        .then(res => res[0])
+        ?? null
+    },
+    getUserByEmail: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.email, data))
+        .then(res => res[0])
+        ?? null
+    },
+    createSession: async (data) => {
+      return client
+        .insert(sessions)
+        .values(data)
+        .returning()
+        .then(res => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return client.select({
+        session: sessions,
+        user: users
+      })
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data))
+        .innerJoin(users, eq(users.id, sessions.userId))
+        .then(res => res[0])
+        ?? null
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      return client
+        .update(users)
+        .set(data)
+        .where(eq(users.id, data.id))
+        .returning()
+        .then(res => res[0])
+    },
+    updateSession: async (data) => {
+      return client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .returning()
+        .then(res => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      const updatedAccount = await client
+        .insert(accounts)
+        .values(rawAccount)
+        .returning()
+        .then(res => res[0])
+
+      const account: ReturnType<Adapter["linkAccount"]> = {
+        ...updatedAccount,
+        access_token: updatedAccount.access_token ?? undefined,
```

Comment:
Could you give this block of code some comments, it feels unnecessary to me 🤔

Reply:
Added comment!
      // Drizzle will return `null` for fields that are not defined.
      // However, the return type is expecting `undefined`.